differ/mutationDiffer.go

```diff
@@ -90,20 +90,39 @@ type MutationDiffer struct {
 	utils           xdcrUtils.UtilsIface
 }
 
+type JsonBody struct {
+	Value    []byte
+	Cas      uint64
+	Datatype uint8
+}
+
+type JsonMeta struct {
+	Value    []byte
+	Cas      uint64
+	Seqno    uint64
+	Flags    uint32
+	Expiry   uint32
+	Deleted  uint32
+	Datatype uint8
+	Cvcas    uint64
+	CvSrc    hlv.DocumentSourceId
+	CvVer    uint64
+	Pv       hlv.VersionsMap
+	Mv       hlv.VersionsMap
+	Updated  bool
+}
+
 // GocbResult is a wrapper struct that is composed with properties for both get and getMeta results from gocb
 type GocbResult struct {
-	*gocbcore.GetResult
-	*gocbcore.GetMetaResult
-	*gocbcore.LookupInResult
+	*JsonBody
+	*JsonMeta
 }
 
 func (r *GocbResult) MarshalJSON() ([]byte, error) {
-	if r.GetResult != nil {
-		return json.Marshal(r.GetResult)
-	} else if r.GetMetaResult != nil {
-		return json.Marshal(r.GetMetaResult)
-	} else if r.LookupInResult != nil {
-		return json.Marshal(r.LookupInResult)
+	if r.JsonBody != nil {
```

Comment:
You said that you wanted a "single way" to do things. Is that what your plan is?
Right now, the marshal method would only either marshal JsonBody *or* jsonMeta. Did you want the marshal to be able handle both?

Reply:
The reason behind me splitting the struct into jsonBody and JsonMeta is when I choose the compare type to be "body only", we should json encode only the body (and maybe a few metadata like CAS and revID). If I retain the single struct while encoding, then we will be encoding info that might not be so useful when the compare type is "body only".
But now I populate the data to be encoded to a map and then JsonMarshal the map